src/passes/OptimizeInstructions.cpp

```diff
@@ -907,6 +907,29 @@ struct OptimizeInstructions
       }
     }
 
+    if (curr->op == ExtendUInt32 || curr->op == ExtendSInt32) {
+      if (auto* load = curr->value->dynCast<Load>()) {
+        // i64.extend_i32_s(i32.load(_8|_16)(_u|_s)(x))  =>
+        //    i64.load(_8|_16|_32)(_u|_s)(x)
+        //
+        // i64.extend_i32_u(i32.load(_8|_16)(_u|_s)(x))  =>
+        //    i64.load(_8|_16|_32)(_u|_s)(x)
+        //
+        // but skip special two special cases:
+        //   i64.extend_i32_u(i32.load8_s(x)) and
+        //   i64.extend_i32_u(i32.load16_s(x))
```

Comment:
Please add an explanation here in the source code - explaining in the PR is not enough, as people will read the code in the future and not the PR (except if they look for it).

Reply:
Sure!